### ISSUE TITLE
"FSharp.Core.sigdata not found..." due to possibly incorrect FSharp.Core 4 Nuget delivery?

### ISSUE THREAD

**Issue by bentayloruk** [fsharp/fsharp#461]
*I've spent some time trying to figure this out, but I'm chasing my tail a little.  Hoping someone here might be able to point me in the right direction!*

Just did a Nuget `update-package FSharp.Core` which took my solution projects from the Nuget `3.1.2.5` to ~~`4.0.0.1`~~ `4.0.0`.  However, all of my F# projects now fail with the following errors:

**Update: it's not going to `4.0.0.1`, it's getting an old package with version `4.0.0.0`.**  *The following information about the `sigdata` looks to be a red herring and my real issue is why I don't get the correct latest FSharp 4 package.*

> FSC: error FS1223: FSharp.Core.sigdata not found alongside FSharp.Core
> FSC: error FS0229: Error opening binary file 'C:\@build\blah\source\packages\FSharp.Core.4.0.0\lib\FSharp.Core.dll': FSharp.Core.sigdata not found alongside FSharp.Core
> FSC: error FS3160: Problem reading assembly 'FSharp.Core, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a': Error opening binary file 'C:\@build\blah\source\packages\FSharp.Core.4.0.0\lib\FSharp.Core.dll': FSharp.Core.sigdata not found alongside FSharp.Core

The projects have the following properties:

* Target Framework = .NET Framework 4.6
* Target F# Runtime = 4.0.0.0 *This is not editable*

I've done the Googling on sigdata, but I don't want to start moving files around, until I understand what I've done wrong.  Anyone have any ideas?  Have I raised this in the correct place?

*Classic Monday morning issue, as I was only doing the update to see if it had any impact on an FsCheck issue I was having! :-) https://github.com/fscheck/FsCheck/issues/124*

**Comment by bentayloruk** [comment]
Not sure what happened, but the `fsharp.core.4.0.0.1\lib\net40` in my `packages` folder, looks nothing like the folder I see if I download the Nuget package and extract it.  Think something must have gone wrong when it installed.  Going to nuke and install again.

**Comment by bentayloruk** [comment]
It seems to me, that I am unable to install [the F# 4.0 package](https://www.nuget.org/packages/FSharp.Core/4.0.0.1).  I'm hoping it's not because of something stupid I'm doing!

In order to rule out something weird in my solution, I just tried to repro in a new Console Application.  I get the same issue, with some additional oddities.  I must be doing something wrong.

I did the following:

* Open VS 2015.
* Create new F# Console Application.
* Set **Target Framework** to `4.6`.  Target F# runtime is F# 4.0 (`FSharp.Core, 4.4.0.0`).
* *Builds and runs OK.  Seems weird that I can't remove the `FSharp.Core reference`.*
* `install-package FSharp.Core` from **Package Manager Console** (PMC).
* It installs `FSharp.Core 3.1.2.5`.  *Shouldn't this be 4?*.
* `upgrade-package FSharp.Core` from PMC.
* PMC installs `FSharp.Core 4.0.0`.
* Build fails with `FSharp.Core.sigdata not found alongside FSharp.Core` etc.

When I look in `\packages\FSharp.Core.4.0.0\lib` it only contains `FSharp.Core.dll` and `FSharp.Core.xml`.  However, when I [download the package](https://www.nuget.org/packages/FSharp.Core/) (using NuTake), the net40 folder contains the `optdata` and the `sigdata`.  It seems I'm getting an old package from somewhere (I've deleted all my local Nuget caches).

The nuspec for the package looks like this, which is clearly wrong and nothing like the nuspec in the version I extracted.

``` xml
<?xml version="1.0"?>
<package xmlns="http://schemas.microsoft.com/packaging/2011/08/nuspec.xsd">
  <metadata>
    <id>FSharp.Core</id>
    <version>4.0.0</version>
    <authors>Microsoft</authors>
    <owners>Microsoft</owners>
    <requireLicenseAcceptance>false</requireLicenseAcceptance>
    <description>FSharp.Core.dll</description>
    <releaseNotes>V 4.0.0</releaseNotes>
    <copyright>Copyright 2012</copyright>
    <tags>f#</tags>
  </metadata>
</package>
```

I've tried installing the specific version `4.0.0.1` but PMC Nuget says `FSharp.Core` is not found.

Any ideas on what stupid thing I'm doing?

**Comment by bentayloruk** [comment]
Sussed it.  This issue appears to be a problem with the V3 Nuget API feed.  My Visual Studio 2015 only had the following package source `https://api.nuget.org/v3/index.json` configured.  When using Package Manager Console and this feed, it was impossible to install the latest F# version.  I kept getting [this old, unlisted version](https://www.nuget.org/packages/FSharp.Core/4.0.0) and only when asking to `update-package`!  When using the PMC `-Version` auto-complete, none of the `4.x` versions are listed.

My workaround:

* Visual Studio > Tools > Options > Nuget Package Manager > Package Sources.
* Add new source by clicking the green plus icon.
* Call it *Old nuget.org*.
* Configure **Source** as `https://www.nuget.org/api/v2`.
* Move it *above* the `api.nuget.org...` entry.
* Click OK.
* Curse the amount of time I lost to this.

After doing this, I was able to install `4.0.0.1`, which is all I ever wanted.

*Paket coming soon! :-)*
